app/src/main/java/org/listenbrainz/android/repository/ArtistRepository.kt

```diff
@@ -8,7 +8,7 @@ import org.listenbrainz.android.model.Song
 interface ArtistRepository {
  fun getArtist(artistID: String) : Flow<Artist>
  fun getArtists(): Flow<List<Artist>>
- suspend fun addArtists(): Boolean
- suspend fun addAllSongsOfArtist(artist: Artist): List<Song>
+ suspend fun addArtists(userRequestedRefresh: Boolean = false): Boolean
+ suspend fun addAllSongsOfArtist(artist: Artist, userRequestedRefresh: Boolean): List<Song>
```

Comment:
Let's look for solutions regarding this later on. Approving for now.

Reply:
I tried doing what you said, but too much unnecessary code boiler plate code was generated. if you want I can document this further.